### PR TITLE
SQL-1065: Unit test json_schema and MongoQuery helper methods

### PR DESCRIPTION
**Pull request by mattChiaravalloti**
This PR adds unit tests for the nontrivial helper functions added in SQL-638 (#19).

Sorry for the delay! There were more hiccups along the road to getting this new laptop working than expected 🤦‍♀️ 

Just tagging you, Patrick, since you were ultimately the only one to review the previous PR and this just adds tests for the work from that one!

### CODE REVIEW

File: core/src/query.rs [hunk 1]
```diff
@@ -250,3 +250,279 @@ impl ObjectSchema {
...
+                        )
+                    }
+                }
+            }
+        }
+
+        // TODO:
+        //  - top-level not Object => error
+        //  - prop not Object => error
+        //  - output sorted alphabetically
```
Comment (pmeredit) [inline core/src/query.rs:376]:
```suggestion
```

**Review by pmeredit — APPROVED** [review]
Just the one little fix, otherwise looks great!

**Comment by pmeredit** [comment]
Oh, and some clippy errors. Clippy probably updated again, haha

File: core/src/json_schema.rs [hunk 2]
```diff
@@ -187,7 +187,14 @@ pub mod simplified {
                         additional_properties: additional_properties.unwrap_or(true),
                     })),
                     BsonType::Single(t) => Ok(Atomic::Scalar(t)),
-                    BsonType::Multiple(_) => Err(Error::InvalidResultSetJsonSchema),
+                    BsonType::Multiple(types) => {
+                        match types[..] {
```
[the remark this answers is not in the document]
Comment (mattChiaravalloti) [reply core/src/json_schema.rs:191]: Oh yea, that's a good idea! I will make that change.

**Comment by mattChiaravalloti** [comment]
evergreen merge

**Pull request merged**